base-ci-resource.yaml

```diff
@@ -0,0 +1,198 @@
+#---shared requirements of git lab ci jobs and stages---
+
+#base compoenets of postgres migration jobs 
+.migrate:
+  image: kevinrambaud/node-flyway:8-5.1.4-alpine
+  script:
+    - cp -r $CI_PROJECT_DIR/sql/* /flyway/sql/
+    - sh /flyway/flyway -url=jdbc:postgresql://$POSTGRES_SERVER:5432/$POSTGRES_DB -schemas=public -user=$POSTGRES_USER -password=$POSTGRES_PASSWORD -placeholders.ENVIRONMENT=$CI_ENVIRONMENT_NAME migrate
+
+# base components of jobs that use node 
+.node: 
+  before_script:
+    - export VERSION=${CI_COMMIT_SHA:0:7}
+    - echo "@cimpress-technology:registry=https://artifactory.cimpress.io/artifactory/api/npm/npm-release-local/" > ~/.npmrc
+    - echo "//artifactory.cimpress.io/artifactory/api/npm/npm-release-local/:_authToken=${CT_ARTIFACTORY_NPM_TOKEN}" >> ~/.npmrc
+    - npm i -g grunt
+    - npm ci
+
+#Base components for setting up a deploy configuration. 
+.deploy:
+  image: briangweber/docker-node:carbon
+  extends: .node
+  script:
+    - grunt deploy --environment=$CI_ENVIRONMENT_NAME --packageVersion=$VERSION --packageName=$CI_PROJECT_NAME
+
+# Default before script 
+before_script:
+  - export VERSION=${CI_COMMIT_SHA:0:7}
+
+#---Build and testing jobs---#
+
+# If project is not a node app it will use this job configuration for build and testing 
+testNodeApplication:
+  extends: .node
+  image: briangweber/docker-node:carbon
+  stage: build  
+  only:
+    variables:       
+      - $NODEPOSTGRESAPP == null  
+  script:
+    - npm test
+  coverage: '/All files\s*\|\s*[.?\d]+\s*\|\s*([0-9.]+)/'
+  artifacts:
+    paths:
+      - coverage/
+
+# if project is a node application using a postgres db this will be the build and test job 
+testNodeApplicationUsingPostgres:
+  extends: .node
+  image: kevinrambaud/node-flyway:8-5.1.4-alpine
+  stage: build  
+  only:
+    variables:       
+      - $NODEPOSTGRESAPP == "true"  
+  script:
+    - cp -r $CI_PROJECT_DIR/sql/* /flyway/sql/
+    - sh /flyway/flyway -url=jdbc:postgresql://$POSTGRES_SERVER:5432/$POSTGRES_DB -schemas=public -user=$POSTGRES_USER -password=$POSTGRES_PASSWORD -placeholders.ENVIRONMENT=$CI_ENVIRONMENT_NAME migrate
```

Comment:
Yes, you swapped to use `CI_ENVIRONMENT_NAME` here, but you don't set it anywhere. I'm not sure it makes sense to set, given that this isn't an environment, you just need to tell flyway what placeholder to set. I'd personally hardcode "dev" instead of using an environment variable at all here